app/models/mixins/container_resource_mixin.rb

```diff
@@ -0,0 +1,101 @@
+# ContainerResourceMixin provides capabilities to container
+# resources.
+module ContainerResourceMixin
+  extend ActiveSupport::Concern
+
+  def spec
+    resource = provider_definition
+    unless resource[:spec].nil?
+      return resource[:spec]
+    end
+    nil
+  end
+
+  def spec=(value)
+    resource = provider_definition
+    unless resource[:spec].nil?
+      resource[:spec] = value
+      response = update_in_provider(resource)
+      return response
+    end
+    nil
+  end
+
+  def annotations(annotation = nil)
+    resource = provider_definition
+    if annotation.nil?
+      return resource[:metadata][:annotations]
+    end
+    resource[:metadata][:annotations][annotation.to_sym]
+  end
+
+  def kind_in_provider
+    MIQ_CLASS_MAPPING[self.class.name]
```

Comment:
@zakiva @cben I agree we should have these both in one place.  Placing it in the Kubernetes provider module makes sense to me.  @zakiva there is no good reason this list is partial.  This subset represents only the entities I thought ought to be affected by the functions in this pull request.  Perhaps a complete list would be better.

Would you like me to open a pull request on the Kubernetes module?  Is there a specific way you'd like to have the inverted list implemented (e.g. static)?